huaweicloud/services/waf/resource_huaweicloud_waf_rule_geolocation_access_control.go

```diff
@@ -0,0 +1,261 @@
+// ---------------------------------------------------------------
+// *** AUTO GENERATED CODE ***
+// @Product WAF
+// ---------------------------------------------------------------
+
+package waf
+
+import (
+	"context"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/jmespath/go-jmespath"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceRuleGeolocation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceRuleGeolocationCreate,
+		UpdateContext: resourceRuleGeolocationUpdate,
+		ReadContext:   resourceRuleGeolocationRead,
+		DeleteContext: resourceRuleGeolocationDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceWAFRuleImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"policy_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Specifies the policy ID.`,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `Specifies the name of WAF geolocation access control rule.`,
+			},
+			"geolocation": {
+				Type:        schema.TypeString,
```

Comment:
how about change the type to set, and convert it th string in the code

Reply:
Make the field type consistent with the document.